fl4health/clients/flash_client.py

```diff
@@ -0,0 +1,139 @@
+from logging import INFO
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Tuple
+
+import torch
+from flwr.common.logger import log
+from flwr.common.typing import Config, Scalar
+
+from fl4health.checkpointing.client_module import ClientCheckpointModule
+from fl4health.clients.basic_client import BasicClient
+from fl4health.utils.losses import LossMeterType
+from fl4health.utils.metrics import Metric
+
+
+class FlashClient(BasicClient):
+    def __init__(
+        self,
+        data_path: Path,
+        metrics: Sequence[Metric],
+        device: torch.device,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        checkpointer: Optional[ClientCheckpointModule] = None,
+    ) -> None:
+        """
+        This client is used to perform client-side training associated with the FLASH method described in
+        https://proceedings.mlr.press/v202/panchal23a/panchal23a.pdf. FLASH is designed to handle statistical
+        heterogeneity and concept drift in federated learning through client-side early stopping and server-side
+        drift-aware adaptive optimization.
+
+        Args:
+            data_path (Path): Path to the data directory.
+            metrics (Sequence[Metric]): List of metrics to be used for evaluation.
+            device (torch.device): Device to be used for training.
+            loss_meter_type (LossMeterType, optional):
+                Type of loss meter to be used. Defaults to LossMeterType.AVERAGE.
+            checkpointer (Optional[ClientCheckpointModule], optional): Checkpointer module defining when and how to do
+                checkpointing during client-side training. No checkpointing is done if not provided. Defaults to None.
+            gamma (float, optional):
```

Comment:
gamma isn't actually an argument to this client. So I'd take this out of the doc string.

Reply:
done in latest commit!